### ISSUE TITLE
Cygwin: Windows absolute paths not recognized as such

### ISSUE THREAD

**Issue by mtzguido** [ocaml-batteries-team/batteries-included#1136]
Hi, I noticed that `is_absolute` does not return true for Windows paths when in a Cygwin environment. For instance the program below prints false, when I would expect a true.

```ocaml
let _ =
  let open BatPathGen.OfString in
  let fn = "C:\\a\\b\\c" in
  let path = of_string fn in
  print_endline (string_of_bool (is_absolute path))
```
I think this is due to this snippet, which simply does not check for `"Cygwin"` too:
https://github.com/ocaml-batteries-team/batteries-included/blob/00c591ea19ab61f01cb77e8ff5e9b5ee4c33ac45/src/batPathGen.ml#L495-L497
Is this intentional or maybe just an oversight...?

This was tested on a fresh Cygwin install with batteries 3.8.0, I can provide more info if needed.

**Comment by UnixJunkie** [comment]
Do you have a proposed fix that you can test?

**Comment by UnixJunkie** [comment]
I.e. a pull request. Not everybody has access to Cygwin.

**Comment by mtzguido** [comment]
Yes I can test and submit a PR today, adding "Cygwin" above does seem to fix it. I'll add some unit tests too but I suppose there's no automated Cygwin test run?

**Comment by UnixJunkie** [comment]
I do not manage batteries' CI tests so I don't know.

**Comment by UnixJunkie** [comment]
If you can implement your proposed fix and test that it works; that's enough for me.

**Comment by mtzguido** [comment]
Sorry for the delay, that does fix this immediate issue but causes some more paths to be printed in Windows format, which I suppose is not desirable. I'm also not well versed in Cygwin at all, trying to make some sense of it before sending a PR here.
